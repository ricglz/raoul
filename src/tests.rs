use super::{parse, DirFunc};

#[test]
fn use_of_undeclared_variable() {
    let filename = "examples/invalid/undeclared-variable.ra";
    let program = std::fs::read_to_string(filename).expect(filename);
    let debug = true;
    let ast_response = parse(&program, debug);
    assert!(ast_response.is_ok());
    let ast = ast_response.unwrap();
    let mut dir_func = DirFunc::new();
    let build_res = dir_func.build_dir_func(ast);
    assert!(build_res.is_err());
}
