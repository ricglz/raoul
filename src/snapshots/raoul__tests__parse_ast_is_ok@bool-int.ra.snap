---
source: src/tests.rs
expression: quad_manager
input_file: src/examples/valid/bool-int.ra
---
0    - Goto       -     -     1
1    - Eq         3000  3750  2750
2    - GotoF      2750  -     5
3    - Print      3000  -     -
4    - PrintNl    -     -     -
5    - Eq         3000  3250  2750
6    - GotoF      2750  -     9
7    - Print      3001  -     -
8    - PrintNl    -     -     -
9    - Eq         3000  3500  2750
10   - GotoF      2750  -     13
11   - Print      3002  -     -
12   - PrintNl    -     -     -
13   - Assignment 3000  -     1000
14   - Assignment 3751  -     1000
15   - Print      1000  -     -
16   - PrintNl    -     -     -
17   - End        -     -     -

